File: solutions/kotlin/sum-of-multiples/1/src/main/kotlin/SumOfMultiples.kt
object SumOfMultiples {

    fun sum(factors: Set<Int>, limit: Int): Int {
        val points = mutableSetOf<Int>()
        for (factor in factors) {
            var i = 1
            while (i * factor < limit && i * factor > 0) {
                points.add(i * factor)
                i += 1
            }
        }
        return points.sum()
    }
}

File: solutions/scala/sum-of-multiples/1/src/main/scala/SumOfMultiples.scala
object SumOfMultiples {
  def sum(factors: Set[Int], limit: Int): Int = {
    val res = (1 until limit).filter(x => factors.exists(x % _ == 0)).reduceOption(_ + _)
    res match {
      case Some(value) => value
      case None => 0
    }
  }
}

